department-of-veterans-affairs/Virtual-Regional-Office
# Houli/htn algorithms

# Background

We are creating a new VRO that includes only a very limited set of functionality. We're calling it the "Mini VRO". It lives in this repo in the `hypertension/dc-7101-algorithms/` directory (which I poorly named, and should have instead named `hypertension/mini-vro`) (but I don't want to do any renames right now because we are in the middle of a lot of changes and PRs).

The Mini VRO will be a single REST API endpoint that run in one Lambda function with one API Gateway.

This PR sets up the Mini VRO infrastructure:
- Python Poetry project
- AWS SAM template.yaml file with a Lambda function
- Pytest setup, including a couple dummy passing tests.

# To Test

Run all the makefile commands. They should all work as expected (e.g. the `poetry-install` command installs your dependencies; the `test` command runs pytest and they pass; the deploy commands successfully deploy the SAM stack to AWS).

After you have run `make deploy-guided`, Log into AWS and test the function using the Lambda console. Your test event must at least have a JSON object in the body, but it doesn't matter what the object is.

The result should be HTTP status 200 with a response body of
```json
{
  "statusCode": 200,
  "body": "test body"
}
```
